#! /bin/sh -e
## 69_add_mips3.dpatch by Ralf Baechle <ralf@linux-mips.org>
##
## DP: Fix damage done by cfbc9cac62f0438cefe171736729e786b45884b8.

#commit 69ccd7750a75abcaf893e2d705dc82e9452376c8
#Author: Ralf Baechle <ralf@linux-mips.org>
#Date:   Mon Jan 23 18:53:35 2006 +0000

#    Fix damage done by cfbc9cac62f0438cefe171736729e786b45884b8.

#    The .set mips3 is needed so the NMI handler builds properly when
#    configured for MIPS I even though no MIPS I processor is using it atm.

#    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>



if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0


diff --git a/arch/mips/kernel/genex.S b/arch/mips/kernel/genex.S
index 1bcbd91..13f22d1 100644
--- a/arch/mips/kernel/genex.S
+++ b/arch/mips/kernel/genex.S
@@ -237,6 +237,7 @@ NESTED(nmi_handler, PT_SIZE, sp)
  	move	a0, sp
 	jal	nmi_exception_handler
 	RESTORE_ALL
+	.set	mips3
 	eret
 	.set	pop
 	END(nmi_handler)

